### PR TITLE
Better default NEP hyperparameters

### PR DESCRIPTION
**Pull request by brucefan1983**
We should expose the *best* default hyperparameters to the users.

* Based on our recent tests, $L_1$ regularization is not needed. $L_1$ regularization can totally kill some parameters, which makes new training restarting from old one(s) difficult. It takes a long time to bring the dead parameters alive.
* Set the default values for $N_{\rm bas}^{\rm R/A}$ from (12, 12) back to (8, 8), which are actually more than enough. Using (12,12) leads to too many trainable parameters for large models with many species.

### CODE REVIEW

**Comment by shdchen** [comment]
Yes, these hyperparameters appear to be optimal default choice according to our recent tests, with a good balance between accuracy and efficiency.

**Comment by elindgren** [comment]
I actually see L1 completely removing parameters as a feature, as it gives more robust models in the sense of being less likely to overfit. 

With starting training from old ones, do you mean transfer learning? So starting the training of one model from another?

**Comment by shdchen** [comment]
> I actually see L1 completely removing parameters as a feature, as it gives more robust models in the sense of being less likely to overfit.
> 
> With starting training from old ones, do you mean transfer learning? So starting the training of one model from another?
> 
> Ping @erhart1 what are your thoughts on this?

Thank you, @elindgren, for sharing your insights.

Yes, our recent tests involved multiple iterations of the NEP model training on various datasets of increasing sizes. We observed that employing L1 regularization during the initial stages might limit the model's adaptability to new datasets in subsequent stages. Therefore, in the context of multi-stage training, temporarily excluding L1 regularization could help maintain the flexibility of the early-stage 'baby' model.

However, L1 regularization could still be valuable for finalizing the NEP model training, as it could help prevent overfitting and may enhance the robustness of the ultimate 'grownup' model.

Considering that most users may not necessarily use multi-stage training, perhaps we could keep L1 regularization enabled by default?

What do you think? @brucefan1983

**Comment by brucefan1983** [comment]
> I actually see L1 completely removing parameters as a feature, as it gives more robust models in the sense of being less likely to overfit.
> 
> With starting training from old ones, do you mean transfer learning? So starting the training of one model from another?
> 
> Ping @erhart1 what are your thoughts on this?

Many users want to restart a training after adding structures. We are also trying to combined two subsystems (such as one for A,B,C,D,D elemets and the other for E,F,G,H,I elemets) to train a bigger one (for A-I elements). In these cases, we found having L1 reguralization enabled in the early stage(s) can slow down the learning process. As @shdchen commented, one can add L1 regularization at the last step if that is intended. 

My major concern is: what is the best default choice that is usually adopted by a new user?

**Comment by brucefan1983** [comment]
> > I actually see L1 completely removing parameters as a feature, as it gives more robust models in the sense of being less likely to overfit.
> > With starting training from old ones, do you mean transfer learning? So starting the training of one model from another?
> > Ping @erhart1 what are your thoughts on this?
> 
> Thank you, @elindgren, for sharing your insights.
> 
> Yes, our recent tests involved multiple iterations of the NEP model training on various datasets of increasing sizes. We observed that employing L1 regularization during the initial stages might limit the model's adaptability to new datasets in subsequent stages. Therefore, in the context of multi-stage training, temporarily excluding L1 regularization could help maintain the flexibility of the early-stage 'baby' model.
> 
> However, L1 regularization could still be valuable for finalizing the NEP model training, as it could help prevent overfitting and may enhance the robustness of the ultimate 'grownup' model.
> 
> Considering that most users may not necessarily use multi-stage training, perhaps we could keep L1 regularization enabled by default?
> 
> What do you think? @brucefan1983

Yes, I agree to enable L1 regularization by default. After all, the cases we tested are not for beginners.

**Comment by brucefan1983** [comment]
I have kept the L1 regularization by default. @elindgren
